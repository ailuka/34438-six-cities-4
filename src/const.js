export const EstateType = {
  APARTMENT: `apartment`,
  HOUSE: `house`,
  HOTEL: `hotel`,
  ROOM: `room`,
};

export const CityName = {
  AMSTERDAM: `Amsterdam`,
  BRUSSELS: `Brussels`,
  COLOGNE: `Cologne`,
  DUSSELDORF: `Dusseldorf`,
  HAMBURG: `Hamburg`,
  PARIS: `Paris`,
};

export const MapClass = {
  MAP_MAIN: `cities__map`,
  MAP_DETAILS: `property__map`,
};

export const OfferClassNamesForPageType = {
  main: {
    article: `cities__place-card`,
    image: `cities__image-wrapper`,
    list: `cities__places-list tabs__content`,
    info: ``,
  },
  details: {
    article: `near-places__card`,
    image: `near-places__image-wrapper`,
    list: `near-places__list`,
    info: ``,
  },
  favorites: {
    article: `favorites__card `,
    image: `favorites__image-wrapper`,
    list: ``,
    info: `favorites__card-info`,
  }
};

export const KeyCodes = {
  ENTER: 13,
  SPACEBAR: 32,
};

export const SortTypes = {
  POPULAR: 1,
  PRICE_LOW_TO_HIGH: 2,
  PRICE_HIGH_TO_LOW: 3,
  TOP_RATED_FIRST: 4,
};

export const AppRoute = {
  ROOM: `/offer/:id`,
  LOGIN: `/login`,
  FAVORITE: `/favorite`,
  ROOT: `/`,
};
