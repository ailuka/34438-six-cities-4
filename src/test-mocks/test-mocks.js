export const offers = [
  {
    id: 1,
    image: `img/room.jpg`,
    photos: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`],
    price: 120,
    rating: 4.1,
    city: {
      name: `Amsterdam`,
      coordinates: [52.3909553943508, 4.85309666406198],
      zoom: 10,
    },
    coordinates: [52.3909553943508, 4.85939666406198],
    zoom: 12,
    name: `Beautiful & luxurious apartment at great location`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century. An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    bedrooms: 3,
    guests: 4,
    equipment: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Coffee machine`, `Dishwasher`, `Towels`, `Baby seat`, `Cabel TV`],
    type: `apartment`,
    host: {
      id: 1,
      avatar: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      status: true,
    },
    isBookmarked: false,
    isPremium: true,
  },
  {
    id: 2,
    image: `img/apartment-02.jpg`,
    photos: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`],
    price: 80,
    rating: 4.2,
    city: {
      name: `Amsterdam`,
      coordinates: [52.3909553943508, 4.85309666406198],
      zoom: 10,
    },
    coordinates: [52.369553943508, 4.87309666406198],
    zoom: 12,
    name: `Wood and stone place`,
    description:
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
      An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    bedrooms: 2,
    guests: 4,
    equipment: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Coffee machine`, `Dishwasher`, `Towels`, `Baby seat`, `Cabel TV`],
    type: `room`,
    host: {
      id: 2,
      avatar: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      status: false,
    },
    isBookmarked: true,
    isPremium: false,
  },
  {
    id: 3,
    image: `img/apartment-03.jpg`,
    photos: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`],
    price: 132,
    rating: 4.3,
    city: {
      name: `Amsterdam`,
      coordinates: [52.3909553943508, 4.85309666406198],
      zoom: 10,
    },
    coordinates: [52.369553943508, 4.87509666406198],
    zoom: 12,
    name: `Canal view Prinsengracht`,
    description:
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
      An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    bedrooms: 3,
    guests: 4,
    equipment: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Coffee machine`, `Dishwasher`, `Towels`, `Baby seat`, `Cabel TV`],
    type: `apartment`,
    host: {
      id: 4,
      avatar: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      status: false,
    },
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 4,
    image: `img/apartment-01.jpg`,
    photos: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`],
    price: 180,
    rating: 5,
    city: {
      name: `Amsterdam`,
      coordinates: [52.3909553943508, 4.85309666406198],
      zoom: 10,
    },
    coordinates: [52.369553943508, 4.95309666406198],
    zoom: 12,
    name: `Nice, cozy, warm big bed apartment`,
    description:
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
      An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    bedrooms: 2,
    guests: 4,
    equipment: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Coffee machine`, `Dishwasher`, `Towels`, `Baby seat`, `Cabel TV`],
    type: `apartment`,
    host: {
      id: 3,
      avatar: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      status: true,
    },
    isBookmarked: false,
    isPremium: true,
  },
  {
    id: 5,
    image: `img/apartment-01.jpg`,
    photos: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`],
    price: 180,
    rating: 5,
    city: {
      name: `Paris`,
      coordinates: [48.8647163943578, 2.359309666406199],
      zoom: 10,
    },
    coordinates: [48.8647163943578, 2.359309666406199],
    zoom: 12,
    name: `Nice, cozy, warm big bed apartment`,
    description:
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
      An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    bedrooms: 2,
    guests: 4,
    equipment: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Coffee machine`, `Dishwasher`, `Towels`, `Baby seat`, `Cabel TV`],
    type: `apartment`,
    host: {
      id: 2,
      avatar: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      status: true,
    },
    isBookmarked: false,
    isPremium: true,
  }
];

export const reviews = [
  {
    id: 1,
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    rating: 4.1,
    user: {
      id: 1,
      name: `Max`,
      status: true,
      avatar: `img/avatar-max.jpg`,
    },
    date: `2019-04-24`,
  },
  {
    id: 2,
    text: `Very good location ,lovely staff ,clean apartment, great beds.`,
    rating: 4.8,
    user: {
      id: 2,
      name: `Anthony`,
      status: false,
      avatar: `img/avatar-max.jpg`,
    },
    date: `2019-06-25`,
  },
  {
    id: 3,
    text: `Neighbours were noisy until the early hours.`,
    rating: 3.8,
    user: {
      id: 3,
      name: `Jerry`,
      status: true,
      avatar: `img/avatar-max.jpg`,
    },
    date: `2019-05-12`,
  },
  {
    id: 4,
    text: `Excellent location. Apartment well laid out and very roomy.`,
    rating: 4.2,
    user: {
      id: 4,
      name: `Thomas`,
      status: true,
      avatar: `img/avatar-max.jpg`,
    },
    date: `2019-06-12`,
  },
  {
    id: 5,
    text: `Location was fantastic, Amenities were good, views were good, would probably stay again. Nice large comfortable penthouse with outside areas to sit. Lounge was comfortable and kitchen had what was needed. Close to shops and supermarkets.`,
    rating: 5.0,
    user: {
      id: 5,
      name: `Larry`,
      status: false,
      avatar: `img/avatar-max.jpg`,
    },
    date: `2019-06-10`,
  },
  {
    id: 6,
    text: `Very spacious, clean and in a perfect location.`,
    rating: 4.9,
    user: {
      id: 6,
      name: `Kim`,
      status: false,
      avatar: `img/avatar-max.jpg`,
    },
    date: `2019-05-11`,
  },
  {
    id: 7,
    text: `Nice, but also the check in was very confusing....especially where the office is actually located......putting the notice on the door might work better than on the window where youâ€™re not looking.`,
    rating: 4.6,
    user: {
      id: 7,
      name: `Holy`,
      status: false,
      avatar: `img/avatar-max.jpg`,
    },
    date: `2019-05-05`,
  },
  {
    id: 8,
    text: `The master bed room is spacious and there should be an additional chair and a desk. the last bed room closet need hangers.`,
    rating: 4.8,
    user: {
      id: 8,
      name: `Dylan`,
      status: false,
      avatar: `img/avatar-max.jpg`,
    },
    date: `2019-06-21`,
  },
  {
    id: 9,
    text: `There was nothing I didn't like, it was far better than what I expected!!`,
    rating: 5.0,
    user: {
      id: 9,
      name: `Bernadette`,
      status: false,
      avatar: `img/avatar-max.jpg`,
    },
    date: `2019-05-22`,
  },
  {
    id: 10,
    text: `Excellent location. Apartment well laid out and very roomy.`,
    rating: 4.2,
    user: {
      id: 10,
      name: `Tanya`,
      status: false,
      avatar: `img/avatar-max.jpg`,
    },
    date: `2019-04-11`,
  },
  {
    id: 11,
    text: `No major issues just minor ones of the TV was not a smart TV, so it could connect to our devices wireless and we had to use a cable to stream things from our laptop.`,
    rating: 4.4,
    user: {
      id: 11,
      name: `Caroline`,
      status: false,
      avatar: `img/avatar-max.jpg`,
    },
    date: `2019-03-10`,
  },
  {
    id: 12,
    text: `Very spacious, clean and in a perfect location.`,
    rating: 4.9,
    user: {
      id: 12,
      name: `Sandra`,
      status: false,
      avatar: `img/avatar-max.jpg`,
    },
    date: `2019-05-11`,
  },
];

export const offersNearby = [
  {
    id: 1,
    image: `img/room.jpg`,
    photos: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`],
    price: 120,
    rating: 4.1,
    city: {
      name: `Amsterdam`,
      coordinates: [52.3909553943508, 4.85309666406198],
      zoom: 10,
    },
    coordinates: [52.3909553943508, 4.85939666406198],
    zoom: 12,
    name: `Beautiful & luxurious apartment at great location`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century. An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    bedrooms: 3,
    guests: 4,
    equipment: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Coffee machine`, `Dishwasher`, `Towels`, `Baby seat`, `Cabel TV`],
    type: `apartment`,
    host: {
      id: 1,
      avatar: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      status: true,
    },
    isBookmarked: false,
    isPremium: true,
  },
  {
    id: 2,
    image: `img/apartment-02.jpg`,
    photos: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`],
    price: 80,
    rating: 4.2,
    city: {
      name: `Amsterdam`,
      coordinates: [52.3909553943508, 4.85309666406198],
      zoom: 10,
    },
    coordinates: [52.369553943508, 4.87309666406198],
    zoom: 12,
    name: `Wood and stone place`,
    description:
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
      An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    bedrooms: 2,
    guests: 4,
    equipment: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Coffee machine`, `Dishwasher`, `Towels`, `Baby seat`, `Cabel TV`],
    type: `room`,
    host: {
      id: 2,
      avatar: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      status: false,
    },
    isBookmarked: true,
    isPremium: false,
  },
  {
    id: 3,
    image: `img/apartment-03.jpg`,
    photos: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`],
    price: 132,
    rating: 4.3,
    city: {
      name: `Amsterdam`,
      coordinates: [52.3909553943508, 4.85309666406198],
      zoom: 10,
    },
    coordinates: [52.369553943508, 4.87509666406198],
    zoom: 12,
    name: `Canal view Prinsengracht`,
    description: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    bedrooms: 3,
    guests: 4,
    equipment: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Coffee machine`, `Dishwasher`, `Towels`, `Baby seat`, `Cabel TV`],
    type: `apartment`,
    host: {
      id: 4,
      avatar: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      status: false,
    },
    isBookmarked: false,
    isPremium: false,
  },
  {
    id: 4,
    image: `img/apartment-01.jpg`,
    photos: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`],
    price: 180,
    rating: 5,
    city: {
      name: `Amsterdam`,
      coordinates: [52.3909553943508, 4.85309666406198],
      zoom: 10,
    },
    coordinates: [52.369553943508, 4.95309666406198],
    zoom: 12,
    name: `Nice, cozy, warm big bed apartment`,
    description: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    bedrooms: 2,
    guests: 4,
    equipment: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`, `Washing machine`, `Coffee machine`, `Dishwasher`, `Towels`, `Baby seat`, `Cabel TV`],
    type: `apartment`,
    host: {
      id: 3,
      avatar: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      status: true,
    },
    isBookmarked: false,
    isPremium: true,
  },
];
