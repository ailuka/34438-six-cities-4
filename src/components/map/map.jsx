import React, {PureComponent, createRef} from "react";
import leaflet from "leaflet";
import {CityCoordinates} from "../../const.js";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {getActiveOffer, getCity} from "../../selectors.js";

const ZOOM = 12;
const iconInactive = leaflet.icon({
  iconUrl: `img/pin.svg`,
  iconSize: [27, 39],
});

const iconActive = leaflet.icon({
  iconUrl: `img/pin-active.svg`,
  iconSize: [27, 39],
});

class Map extends PureComponent {
  constructor(props) {
    super(props);

    this._map = null;
    this._markers = null;
    this._mapRef = createRef();
  }

  _addMarkers(map, activeCard) {
    const {offers} = this.props;
    this._markers = leaflet.layerGroup();
    let icon = iconInactive;

    offers.forEach((offer) => {
      if (activeCard) {
        icon = (offer.id === this.props.activeCard.id) ? iconActive : iconInactive;
      }

      leaflet
        .marker(offer.coordinates, {icon})
        .addTo(this._markers);
    });

    this._markers.addTo(map);
  }

  _clearMarkers() {
    this._markers.clearLayers();
  }

  _createMap(mapRef, city) {
    this._map = leaflet.map(mapRef, {
      center: CityCoordinates[city],
      zoom: ZOOM,
      zoomControl: false,
      marker: true,
    });
  }

  _setMapView(city) {
    this._map.setView(CityCoordinates[city], ZOOM);
  }

  _addTileLayer(map) {
    leaflet
      .tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
        attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
      })
      .addTo(map);
  }

  componentDidMount() {
    const mapRef = this._mapRef.current;
    const {city} = this.props;

    this._createMap(mapRef, city);
    this._setMapView(city);
    this._addTileLayer(this._map);
    this._addMarkers(this._map);
  }

  render() {
    return (
      <div id="map" style={{height: `100%`}} ref={this._mapRef}></div>
    );
  }

  componentDidUpdate(prevProps) {
    if (prevProps.city !== this.props.city) {
      this._clearMarkers();
      this._setMapView(this.props.city);
      this._addMarkers(this._map);
    }

    if (prevProps.activeCard !== this.props.activeCard) {
      this._clearMarkers();
      this._addMarkers(this._map, this.props.activeCard);
    }
  }
}

Map.propTypes = {
  city: PropTypes.string.isRequired,
  offers: PropTypes.arrayOf(
      PropTypes.shape({
        city: PropTypes.string.isRequired,
        coordinates: PropTypes.arrayOf(
            PropTypes.number
        ).isRequired,
      }).isRequired
  ).isRequired,
  activeCard: PropTypes.shape({
    id: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    isPremium: PropTypes.bool.isRequired,
  }),
};

const mapStateToProps = (state) => ({
  city: getCity(state),
  activeCard: getActiveOffer(state),
});

export {Map};
export default connect(mapStateToProps, null)(Map);
