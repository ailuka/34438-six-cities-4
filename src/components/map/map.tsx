import * as React from "react";
import * as leaflet from "leaflet";
import {connect} from "react-redux";
import {getActiveOffer, getCity, getCityDetails} from "../../reducer/filters/selectors";
import {MapType} from "../../const";
import {Offer, CityDetails} from "../../types";

interface Props {
  city: string;
  offers: Offer[];
  activeCard: Offer;
  mapType: string;
  cityDetails: CityDetails;
}

const iconInactive = leaflet.icon({
  iconUrl: `/img/pin.svg`,
  iconSize: [27, 39],
});

const iconActive = leaflet.icon({
  iconUrl: `/img/pin-active.svg`,
  iconSize: [27, 39],
});

class Map extends React.Component<Props> {
  private _mapRef: React.RefObject<HTMLDivElement>;
  private _map = leaflet.Map;
  private _markers = leaflet.LayerGroup;

  constructor(props) {
    super(props);

    this._mapRef = React.createRef();
  }

  shouldComponentUpdate(nextProps) {
    if (this.props.city !== nextProps.city) {
      return true;
    }

    if (!this.props.activeCard && nextProps.activeCard) {
      return true;
    }

    if (this.props.activeCard && !nextProps.activeCard) {
      return true;
    }

    if (this.props.activeCard && nextProps.activeCard) {
      if (this.props.activeCard.id !== nextProps.activeCard.id) {
        return true;
      }
    }

    const currentOffers = this.props.offers;
    const newOffers = nextProps.offers;

    if (currentOffers.length !== newOffers.length) {
      return true;
    } else {
      for (let index = 0; index < currentOffers.length; index++) {
        if (currentOffers[index].id !== newOffers[index].id) {
          return true;
        }
      }
    }

    return false;
  }

  componentDidMount() {
    this._createMap(this._mapRef.current);
    this._setMapView();
    this._addTileLayer();
    this._addMarkers();
  }

  componentDidUpdate(prevProps) {
    this._createMap(this._mapRef.current);
    if (prevProps.city !== this.props.city) {
      this._clearMarkers();
      this._setMapView();
      this._addMarkers();
    } else {
      this._clearMarkers();
      this._addMarkers();
    }
  }

  _addMarkers() {
    if (!this._map) {
      return;
    }

    const {offers} = this.props;
    this._markers = leaflet.layerGroup();
    let icon = iconInactive;

    offers.forEach((offer) => {
      if (this.props.activeCard) {
        icon = (offer.id === this.props.activeCard.id) ? iconActive : iconInactive;
      }

      leaflet
        .marker(offer.coordinates, {icon})
        .addTo(this._markers);
    });

    this._markers.addTo(this._map);
  }

  _clearMarkers() {
    if (this._markers) {
      this._markers.clearLayers();
    }
  }

  _createMap(mapRef) {
    if (!this.props.cityDetails) {
      return;
    }

    if (this._map) {
      return;
    }

    const {coordinates, zoom} = this.props.cityDetails;

    this._map = leaflet.map(mapRef, {
      center: coordinates,
      zoom,
      zoomControl: false,
      marker: true,
    });
  }

  _setMapView() {
    if (!this.props.cityDetails) {
      return;
    }

    if (!this._map) {
      return;
    }

    const {coordinates, zoom} = this.props.cityDetails;
    this._map.setView(coordinates, zoom);
  }

  _addTileLayer() {
    if (!this._map) {
      return;
    }

    leaflet
      .tileLayer(`https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`, {
        attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
      })
      .addTo(this._map);
  }

  render() {
    const {mapType} = this.props;

    let mapClass = ``;
    if (mapType === MapType.MAIN) {
      mapClass = `cities__map`;
    } else if (mapType === MapType.DETAILS) {
      mapClass = `property__map`;
    }

    return (
      <section className={`${mapClass} map`}>
        <div id="map" style={{height: `100%`}} ref={this._mapRef}></div>
      </section>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  city: getCity(state),
  cityDetails: getCityDetails(state),
  activeCard: ownProps.activeCard || getActiveOffer(state),
});

export {Map};
export default connect(mapStateToProps, null)(Map);
