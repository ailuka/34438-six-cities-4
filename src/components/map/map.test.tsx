import React from "react";
import Enzyme, {mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {offers} from "../../test-mocks/test-mocks.js";
import {CityName, MapType} from "../../const.js";
import {Map} from "./map.jsx";

Enzyme.configure({
  adapter: new Adapter(),
});

const cityDetails = {
  name: `Amsterdam`,
  coordinates: [52.3909553943508, 4.85309666406198],
  zoom: 10,
};

describe(`Map snapshot`, () => {
  describe(`map is on the main page`, () => {
    it(`should render with a correct class`, () => {
      const div = global.document.createElement(`div`);
      global.document.body.appendChild(div);

      const tree = mount(
          <Map
            city={CityName.AMSTERDAM}
            offers={offers}
            activeCard={offers[0]}
            mapType={MapType.MAIN}
            cityDetails={cityDetails}
          />,
          {attachTo: div}
      );

      expect(tree.getDOMNode()).toMatchSnapshot();
    });
  });

  describe(`map is on the offer details page`, () => {
    it(`should render with a correct class`, () => {
      const div = global.document.createElement(`div`);
      global.document.body.appendChild(div);

      const tree = mount(
          <Map
            city={CityName.AMSTERDAM}
            offers={offers}
            activeCard={offers[0]}
            mapType={MapType.DETAILS}
            cityDetails={cityDetails}
          />,
          {attachTo: div}
      );

      expect(tree.getDOMNode()).toMatchSnapshot();
    });
  });
});
