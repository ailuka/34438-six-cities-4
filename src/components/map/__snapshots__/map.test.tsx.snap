// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map snapshot map is on the main page should render with a correct class 1`] = `
<section
  class="cities__map map"
>
  <div
    class="leaflet-container leaflet-grab leaflet-touch-drag"
    id="map"
    style="height: 100%; position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      >
        <div
          class="leaflet-layer "
          style="z-index: 1;"
        >
          <div
            class="leaflet-tile-container leaflet-zoom-animated"
            style="z-index: 18; left: 0px; top: 0px;"
          >
            <img
              alt=""
              class="leaflet-tile"
              role="presentation"
              src="https://a.basemaps.cartocdn.com/rastertiles/voyager/10/525/336.png"
              style="width: 256px; height: 256px; left: -206px; top: -110px;"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      >
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin-active.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 5px; top: 0px; z-index: 0;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 14px; top: 25px; z-index: 25;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 16px; top: 25px; z-index: 25;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 73px; top: 25px; z-index: 25;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -1816px; top: 4050px; z-index: 4050;"
          tabindex="0"
        />
      </div>
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      />
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JS library for interactive maps"
          >
            Leaflet
          </a>
           | © 
          <a
            href="https://www.openstreetmap.org/copyright"
          >
            OpenStreetMap
          </a>
           contributors © 
          <a
            href="https://carto.com/attributions"
          >
            CARTO
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Map snapshot map is on the offer details page should render with a correct class 1`] = `
<section
  class="property__map map"
>
  <div
    class="leaflet-container leaflet-grab leaflet-touch-drag"
    id="map"
    style="height: 100%; position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      >
        <div
          class="leaflet-layer "
          style="z-index: 1;"
        >
          <div
            class="leaflet-tile-container leaflet-zoom-animated"
            style="z-index: 18; left: 0px; top: 0px;"
          >
            <img
              alt=""
              class="leaflet-tile"
              role="presentation"
              src="https://a.basemaps.cartocdn.com/rastertiles/voyager/10/525/336.png"
              style="width: 256px; height: 256px; left: -206px; top: -110px;"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      >
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin-active.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 5px; top: 0px; z-index: 0;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 14px; top: 25px; z-index: 25;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 16px; top: 25px; z-index: 25;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 73px; top: 25px; z-index: 25;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="/img/pin.svg"
          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -1816px; top: 4050px; z-index: 4050;"
          tabindex="0"
        />
      </div>
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      />
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JS library for interactive maps"
          >
            Leaflet
          </a>
           | © 
          <a
            href="https://www.openstreetmap.org/copyright"
          >
            OpenStreetMap
          </a>
           contributors © 
          <a
            href="https://carto.com/attributions"
          >
            CARTO
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
